compiler.ts

```diff
@@ -24,13 +24,15 @@ export type GlobalEnv = {
   globals: Map<string, number>;
   classes: Map<string, Map<string, [number, Literal]>>;
   locals: Map<string, number>;      // Map from local/param to stack slot index
+  offset: number;
```

Comment:
Do you guys need `offset` for anything? Otherwise, you should be using `staticAlloc()` Or is this old code that's being merged?